src/cli/insert.rs

```diff
@@ -0,0 +1,105 @@
+//! The `kv-upsert` command performs a KV upsert operation.
+
+use super::util::{json_rows_from_input_columns, json_rows_from_input_optionals};
+
+use crate::state::State;
+use couchbase::InsertOptions;
+
+use futures::executor::block_on;
+use log::debug;
+use nu_cli::{CommandArgs, CommandRegistry, OutputStream};
+use nu_errors::ShellError;
+use nu_protocol::{Signature, SyntaxShape, TaggedDictBuilder};
+use nu_source::Tag;
+use std::sync::Arc;
+
+pub struct Insert {
+    state: Arc<State>,
+}
+
+impl Insert {
+    pub fn new(state: Arc<State>) -> Self {
+        Self { state }
+    }
+}
+
+impl nu_cli::WholeStreamCommand for Insert {
+    fn name(&self) -> &str {
+        "kv-insert"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("kv-insert")
+            .optional("id", SyntaxShape::String, "the document id")
+            .optional("content", SyntaxShape::String, "the document content")
+            .named(
+                "id-column",
+                SyntaxShape::String,
+                "the name of the id column if used with an input stream",
+                None,
+            )
+            .named(
+                "content-column",
+                SyntaxShape::String,
+                "the name of the content column if used with an input stream",
+                None,
+            )
+    }
+
+    fn usage(&self) -> &str {
+        "Insert a document through Key/Value"
+    }
+
+    fn run(
+        &self,
+        args: CommandArgs,
+        registry: &CommandRegistry,
+    ) -> Result<OutputStream, ShellError> {
+        block_on(run_insert(self.state.clone(), args, registry))
+    }
+}
+
+async fn run_insert(
+    state: Arc<State>,
+    args: CommandArgs,
+    registry: &CommandRegistry,
+) -> Result<OutputStream, ShellError> {
+    let mut args = args.evaluate_once(registry)?;
+
+    let id_column = args
+        .get("id-column")
+        .map(|id| id.as_string().unwrap())
+        .unwrap_or_else(|| String::from("id"));
+
+    let content_column = args
+        .get("content-column")
+        .map(|content| content.as_string().unwrap())
+        .unwrap_or_else(|| String::from("content"));
+
+    let mut rows = json_rows_from_input_columns(&mut args, &id_column, &content_column).await?;
```

Comment:
I'm not totally confident that id_column and content_column need to be borrows but it might make sense in other call cases?

Reply:
if it compiles it should be fine, good rule of thumb with rust ;) ..